libtock/services/alarm.h

```diff
@@ -47,38 +47,50 @@ typedef struct alarm {
 
 /** \brief Opaque handle to a repeating alarm.
  *
- * An opaque handle to a repeating alarm created by `libtock_alarm_repeating_every`.
+ * An opaque handle to an alarm created by `libtock_alarm_repeating_every_ms`
+ * and `libtock_alarm_in_ms`.
  */
-typedef struct alarm_repeating {
-  uint32_t interval;
-  libtock_alarm_callback cb;
-  void* ud;
+typedef struct alarm_data {
+  // Length of timer in milliseconds.
+  uint32_t interval_ms;
+  // Number of times the underlying counter will overflow
+  // before hitting the target interval_ms.
+  uint32_t overflows_left;
+  // Number of ticks remaining after the last time the
+  // counter overflows.
+  uint32_t remaining_ticks;
+  libtock_alarm_callback callback;
+  void* user_data;
   libtock_alarm_t alarm;
-} libtock_alarm_repeating_t;
+} libtock_alarm_data_t;
```

Comment:
Why does this modify `libtock_alarm_repeating_t` and not `libtock_alarm_t`? Couldn't a single alarm also have an overflow?

Reply:
`libtock_alarm_at` sets a single alarm and takes in a 32 bit argument (`dt`) for the length of the alarm in ticks. This cannot overflow as the kernel guarantees that the alarm buffer will not overflow sooner than 32 bits of ticks. Only alarms longer than $2^{32} - 1$ ticks can overflow. These alarms can only be set by `libtock_alarm_in_ms`.

Reply:
Ok that makes sense.

What about renaming `libtock_alarm_t` to something like `libtock_alarm_dt_t`, creating a new `libtock_alarm_t` that supports overflows, and leaving `libtock_alarm_repeating_t` as it is? It would be nice to not have to change all of the examples and to keep the more descriptive "repeating" name. Changing the type for `libtock_alarm_at()` seems the least invasive since it isn't really used.

Reply:
I have restored the `libtock_alarm_t` name to make the renaming less invasive. We cannot keep the `libtock_alarm_repeating_t` unchanged for a few reasons:
- `libtock_alarm_repeating_t` sets repeating alarms in milliseconds, so it must support overflows. Thus we must add extra fields to support the new overflow logic.
-  `libtock_alarm_repeating_every_ms` calls `libtock_alarm_in_ms` with the alarm struct it was given. To support the less intrusive naming, both must be the same type name `libtock_alarm_t`. As the struct `libtock_alarm_repeating_t` is not relevant to the user calling these functions, I do not think it is more helpful to include the "repeating" moniker. This is why I believe there should only be two structs and why the name of the `repeating` struct must be changed.

Reply:
Ok makes sense I agree.